projects/doc-lib/src/vanilla/tabset/tabset.component.ts

```diff
@@ -0,0 +1,145 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import {html, render, TemplateResult} from 'lit-html';
+import {JsComponent} from '../js.component';
+
+export interface VcdTab {
+    /**
+     * The tab's title, what appears at the top.
+     * TODO: Consider making this support TemplateResult
+     */
+    title: string | TemplateResult;
+    /**
+     * The tab contents, an HTML string. This is prone to script injection
+     * TODO: Accept a TemplateResult to mitigate this
+     */
+    panel: string | TemplateResult;
+}
+
+interface VcdTabInternal extends VcdTab {
+    /**
+     * Unique ID for this tab, needed for a11y but we don't force users to create it
+     */
+    id: string;
+}
+
+/**
+ * Create unique IDs for tab's aria attributes.
+ */
+const getId = (() => {
```

Comment:
Use IdGenerator in `../vmware-cloud-director-ui-components/projects/components/src/utils/id-generator/id-generator.ts`

Reply:
Actually, it's too small a function to add a dependency on `@vcd/ui-components`